ci/build_docs.sh

```diff
@@ -23,14 +26,11 @@ PYTHON_CHANNEL=$(rapids-download-conda-from-s3 python)
 rapids-mamba-retry install \
   --channel "${CPP_CHANNEL}" \
   --channel "${PYTHON_CHANNEL}" \
-  libraft \
-  libraft-headers \
-  pylibraft \
-  raft-dask
+  "libraft=${RAPIDS_VERSION_MAJOR_MINOR}" \
+  "libraft-headers=${RAPIDS_VERSION_MAJOR_MINOR}" \
+  "pylibraft=${RAPIDS_VERSION_MAJOR_MINOR}" \
+  "raft-dask=${RAPIDS_VERSION_MAJOR_MINOR}"
 
-export RAPIDS_VERSION="$(rapids-version)"
```

Comment:
Why has `RAPIDS_VERSION` been removed?

Reply:
As far as I can tell, it's unnecessary here. It doesn't affect anything from `gha-tools` or anything else invoked in this script.

I suspected that it was copied from conda-building scripts.

Reply:
Hey @KyleFromNVIDIA , I just requested another review because I pushed more changes.... could you look at the new, even stricter form I'm proposing here now?

Based on https://github.com/rapidsai/build-planning/issues/106#issuecomment-2402728240

* using the output of `rapids-version` instead of `rapids-version-major-minor` in constraints, to avoid falling back to older nightlies in the same major release
* setting `RAPIDS_VERSION_NUMBER` as a shell variable when running the one command that needs it (`rapids-upload-docs`) to make it clearer how it's used